packages/devtools-client-adapters/src/types.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+/**
+ * These are intended to be generic types used by Firefox, Chrome, Node, and
+ * possibly others.
+ *
+ * They can be imported into clients by using the following Flow import:
+ * `import type { Location } from 'devtools-client-adapters'`
+ */
+
+/**
+ * Breakpoint ID
+ *
+ * @memberof types
+ * @static
+ */
+export type BreakpointId = string;
+
+/**
+ * Source ID
+ *
+ * @memberof types
+ * @static
+ */
+export type SourceId = string;
+
+/**
+ * Actor ID
+ *
+ * @memberof types
+ * @static
+ */
+export type ActorId = string;
+
+/**
+ * Source File Location
+ *
+ * @memberof types
+ * @static
+ */
+export type Location = {
+  sourceId: SourceId,
+  line: number,
+  column?: number
+};
+
+/**
+ * Breakpoint
+ *
+ * @memberof types
+ * @static
+ */
+export type Breakpoint = {
+  id: BreakpointId,
+  location: Location,
+  loading: boolean,
+  disabled: boolean,
+  text: string,
+  condition: ?string,
+};
+
+/**
+ * Breakpoint Result is the return from an add/modify Breakpoint request
+ *
+ * @memberof types
+ * @static
+ */
+export type BreakpointResult = {
```

Comment:
This is fairly Firefox specific, but also seems to be used in other places so I kept it here for now.  I wanted to have these types become much more of a general case, losing the term `actor` and moving toward `id` to bring Chrome and Firefox together; but that should be done when working on the Chrome side of things.